src/typeInference/annotateTypeSwitchOperator.ts

```diff
@@ -1,7 +1,63 @@
-import { SequenceType } from '../expressions/dataTypes/Value';
-import { IAST } from '../parsing/astHelper';
+import {
+	SequenceMultiplicity,
+	SequenceType,
+	stringToSequenceMultiplicity,
+	stringToValueType,
+} from '../expressions/dataTypes/Value';
+import astHelper, { IAST } from '../parsing/astHelper';
 
 // TODO: Annotation not yet implemented. The docs don't mention it so we don't know the return types.
-export function annotateTypeSwitchOperator(ast: IAST): SequenceType {
+export function annotateTypeSwitchOperator(
+	ast: IAST,
+	argumentType: SequenceType,
+	caseClausesReturns: SequenceType[],
+	defaultCaseReturn: SequenceType
+): SequenceType {
+	const caseClausesConditions = astHelper.getChildren(ast, 'typeswitchExprCaseClause');
+	// Do the switch case to see what will be returned
+	for (let i = 0; i < caseClausesReturns.length; i++) {
+		// The typeswitch either contains a sequenceType with a single type or an 'or' statement
+		if (caseClausesConditions[i][1][0] === 'sequenceType') {
```

Comment:
Really small remark but maybe this can be a switch? Might be a little bit cleaner.